packages/hoppscotch-app/components/environments/index.vue

```diff
@@ -120,65 +120,63 @@
   </div>
 </template>
 
-<script lang="ts">
-import { computed, defineComponent } from "@nuxtjs/composition-api"
-import { useReadonlyStream, useStream } from "~/helpers/utils/composables"
+<script setup lang="ts">
+import { computed, ref } from "@nuxtjs/composition-api"
+import {
+  useReadonlyStream,
+  useStream,
+  useI18n,
+} from "~/helpers/utils/composables"
 import {
   environments$,
   setCurrentEnvironment,
   selectedEnvIndex$,
   globalEnv$,
 } from "~/newstore/environments"
 
-export default defineComponent({
-  setup() {
-    const globalEnv = useReadonlyStream(globalEnv$, [])
+const t = useI18n()
+
+const options = ref<any | null>(null)
+
+const globalEnv = useReadonlyStream(globalEnv$, [])
+
+const globalEnvironment = computed(() => ({
+  name: "Global",
+  variables: globalEnv.value,
+}))
+
+const environments = useReadonlyStream(environments$, [])
+
+const selectedEnvironmentIndex = useStream(
+  selectedEnvIndex$,
+  -1,
+  setCurrentEnvironment
+)
 
-    const globalEnvironment = computed(() => ({
-      name: "Global",
-      variables: globalEnv.value,
-    }))
+const showModalImportExport = ref(false)
+const showModalDetails = ref(false)
+const action = ref("new")
+const editingEnvironmentIndex = ref<string | number | undefined>("Global")
```

Comment:
Only a single string value is possible for the type. "Global" . This can be represented by using TypeScript's String Literal Types
(see comment in `components/environments/Environment.vue` for further explanation)

ALSO, are `environmentIndex` values of type `"Global" | number | undefined` or `"Global" | number | null`. In other places it is marked as `null` see what is the actual value flowing and correct to that type.
undefined is not the same as null in JavaScript and TypeScript upholds that.

Reply:
In the old file, the value was "Global" | number | undefined but in `components/environments/Details.vue` it is null I think null will be more accurate in this case.